pkg/api/dashboard_snapshot_test.go

```diff
@@ -47,13 +47,26 @@ func TestDashboardSnapshotApiEndpoint(t *testing.T) {
 
 		Convey("When user has editor role and is not in the ACL", func() {
 			Convey("Should not be able to delete snapshot", func() {
-				loggedInUserScenarioWithRole("When calling GET on", "GET", "/api/snapshots-delete/12345", "/api/snapshots-delete/:key", m.ROLE_EDITOR, func(sc *scenarioContext) {
+				loggedInUserScenarioWithRole("When calling DELETE on", "DELETE", "/api/snapshots/12345", "/api/snapshots/:key", m.ROLE_EDITOR, func(sc *scenarioContext) {
 					sc.handlerFunc = DeleteDashboardSnapshot
-					sc.fakeReqWithParams("GET", sc.url, map[string]string{"key": "12345"}).exec()
+					sc.fakeReqWithParams("DELETE", sc.url, map[string]string{"key": "12345"}).exec()
 
 					So(sc.resp.Code, ShouldEqual, 403)
 				})
 			})
+
+			Convey("Should be able to delete snapshot by deleteKey", func() {
+				loggedInUserScenarioWithRole("When calling GET on", "GET", "/api/snapshots-delete/12345", "/api/snapshots-delete/:deleteKey", m.ROLE_EDITOR, func(sc *scenarioContext) {
```

Comment:
since this does not require editor I think we should change the test to use `m.ROLE_VIEWER`

Reply:
The new DELETE endpoint requires exactly the same permissions as the GET endpoint does in 5.1.3, is there something I'm missing here?

Reply:
Then we need to add `reqEditorRole` as a middleware to that route. https://github.com/grafana/grafana/pull/12025/files#diff-13d216b12482e706601d65bcc87deb38R110

as in 5.1.3 https://github.com/grafana/grafana/blob/v5.1.x/pkg/api/api.go#L110

Reply:
updated it to use a scenario with no user or role info